app/src/main/java/com/appunite/loudius/ui/login/LoginScreenViewModel.kt

```diff
@@ -0,0 +1,72 @@
+package com.appunite.loudius.ui.login
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.ViewModel
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+
+sealed class LoginAction {
+    object ClearNavigation : LoginAction()
+    object ClickLogIn : LoginAction()
+    object XiaomiPermissionDialogDismiss : LoginAction()
+    object XiaomiPermissionDialogGrantPermission : LoginAction()
+    object XiaomiPermissionDialogAlreadyGrantedPermission : LoginAction()
+}
+
+sealed class LoginNavigateTo {
+    object OpenXiaomiPermissionManager : LoginNavigateTo()
+    object OpenGithubAuth : LoginNavigateTo()
+}
+
+data class LoginState(
+    val showXiaomiPermissionDialog: Boolean,
+    val navigateTo: LoginNavigateTo?,
+)
+
+@HiltViewModel
+class LoginScreenViewModel @Inject constructor(
+    private val githubHelper: GithubHelper,
+) : ViewModel() {
+
+    var state by mutableStateOf(LoginState(showXiaomiPermissionDialog = false, navigateTo = null))
+        private set
+
+    fun onAction(action: LoginAction) {
+        when (action) {
+            LoginAction.ClearNavigation -> {
+                state = state.copy(navigateTo = null)
+            }
+
```

Comment:
question: okay, now I see that you make everywhere these new lines in `when`s. Do you prefer it? I haven't seen this style in kotlin tbh :D

Reply:
I think this was done automatically :)